x-pack/plugins/apm/server/routes/settings/agent_configuration/list_configurations.ts

```diff
@@ -6,27 +6,33 @@
  */
 
 import { APMIndices } from '@kbn/apm-data-access-plugin/server';
+import { APMEventClient } from '../../../lib/helpers/create_es_client/create_apm_event_client';
 import { AgentConfiguration } from '../../../../common/agent_configuration/configuration_types';
 import { convertConfigSettingsToString } from './convert_settings_to_string';
-import { getConfigsAppliedToAgentsThroughFleet } from './get_config_applied_to_agent_through_fleet';
+import { getAgentConfigEtagMetrics } from './get_agent_config_etag_metrics';
 import { APMInternalESClient } from '../../../lib/helpers/create_es_client/create_internal_es_client';
 import { APM_AGENT_CONFIGURATION_INDEX } from '../apm_indices/apm_system_index_constants';
 
-export async function listConfigurations(
-  internalESClient: APMInternalESClient,
-  apmIndices: APMIndices
-) {
+export async function listConfigurations({
+  internalESClient,
+  apmEventClient,
+  apmIndices,
+}: {
+  internalESClient: APMInternalESClient;
+  apmEventClient?: APMEventClient;
+  apmIndices: APMIndices;
+}) {
   const params = {
     index: APM_AGENT_CONFIGURATION_INDEX,
     size: 200,
   };
 
-  const [agentConfigs, configsAppliedToAgentsThroughFleet] = await Promise.all([
+  const [agentConfigs, appliedEtags = []] = await Promise.all([
     internalESClient.search<AgentConfiguration>(
       'list_agent_configuration',
       params
     ),
-    getConfigsAppliedToAgentsThroughFleet(internalESClient, apmIndices),
+    apmEventClient ? getAgentConfigEtagMetrics(apmEventClient) : undefined,
```

Comment:
Can't you call `getIsAppliedByAgent` so the logic is in one place?

Reply:
No, `getIsAppliedByAgent` is for a single agent configuration. In this case we need to fetch every agent configuration, and map them with all available etags so we want to fetch those two in parallel.